code/modules/mob/living/carbon/human/life.dm

```diff
@@ -44,11 +44,16 @@
 
 
 /mob/living/carbon/human/calculate_affecting_pressure(pressure)
-	if (wear_suit && head && istype(wear_suit, /obj/item/clothing) && istype(head, /obj/item/clothing))
```

Comment:
couldn't you just do a 
var/check_this_one = null
if ((wear_suit &| w_uniform) && head) INCLUDE THE ISTYPING HERE
     if(wear_suit.clothing_flags == STOPPRESSUREDAMAGE)
            check_this_one = wear_suit
      else if(same as above but for w_uniform)
      else return pressure
then do the helmet check? AKA slightly less on the processing